RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -134,7 +134,7 @@ void MahiFit::doFit(std::array<float,3> &correctedOutput, int nbx) const {
   }
   else {
     bxSize = bxSizeConf_;
-    nnlsWork_.bxOffset = bxOffsetConf_;
+    nnlsWork_.bxOffset = bxOffsetConf_ + ((static_cast<int>(nnlsWork_.tsOffset) + activeBXs_[0]) >= 0 ? 0 : (nnlsWork_.tsOffset + activeBXs_[0]));
```

Comment:
Considering  how many times this had to be fixed, perhaps there is a problem in the chosen expression complexity.

originally, only ```bxOffsetConf_``` is used and in all configurations made so far it should be equal to ```-activeBXs_[0]``` or (incidentally?) ```abs(activeBXs_[0])``` . So, the result is apparently either the old value (for SOI>=abs(activeBXs_[0]) ) or is equal to SOI.
Is ```activeBXs_[0]``` used here with some hidden meaning, or is it really just supposed to say ```-bxOffsetConf_``` ?

shouldn't it be simpler to use a direct statement 
```c++
nnlsWork_.bxOffset = static_cast<int>(nnlsWork_.tsOffset) >= bxOffsetConf_ ? bxOffsetConf_ : nnlsWork_.tsOffset;

```

Reply:
Hi Slava, Thank you for your comments.

I agree with you. 
First yes "the result is apparently either the old value (for SOI>=abs(activeBXs_[0]) ) or is equal to SOI."
Second, "activeBXs_[0]" doesn't have any hidden meaning. Actually from the current code, [line 35](https://github.com/cms-sw/cmssw/blob/ca6b31c86b25698ee5127c38e885a3528193c0f6/RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc#L35) :
 bxOffsetConf_ = -(*std::min_element(activeBXs_.begin(), activeBXs_.end()));
It's already clear that bxOffsetConf_ is defined from activeBXs_, usually as the order is -activeBXs_[0].

So I think your correction is better both in terms of simpler and also avoiding problems that (possible) disorder of activeBXs may bring. And I think of the logic, it stays unchanged/feasible even we do cases as weird as [0,1,2,3] or [-3,-1,0,1,3] etc.